var/spack/repos/builtin/packages/openmpi/package.py

```diff
@@ -279,6 +281,20 @@ def with_or_without_mxm(self, activated):
             line += '={0}'.format(path)
         return line
 
+    def with_or_without_slurm(self, activated):
+        # PMI and SLURM
+        # https://www.open-mpi.org/faq/?category=slurm
+        # https://slurm.schedmd.com/mpi_guide.html#open_mpi
+        # TODO: Add PMIx support for openmpi(>2.0.0)
+        spec = self.spec
+        if not activated:
+            return '--without-slurm'
+        elif spec.satisfies('@1.5.4:') and '+pmi' not in spec:
+            raise SpackError(
+                "+pmi is required for openmpi(>=1.5.5) to work with SLURM.")
```

Comment:
Can `pmi` be used without slurm? If so I think making it a variant is fine, otherwise we can just pass `--with-mpi` here and avoid the variant. In any case we can move this check above and make it a conflict:
```python
conflicts('schedulers=slurm ~pmi', when='@1.5.4:', msg='+pmi is required for openmpi(>=1.5.5) to work with SLURM.')
```

Reply:
It seems PMI hasn't been adopted in schedulers other than SLURM. So I replaced pmi checking with the conflict in my latest commit.